internal/runners/history/history.go

```diff
@@ -54,22 +54,27 @@ func (h *History) Run(params *HistoryParams) error {
 		if err != nil {
 			return locale.WrapError(err, "err_commit_history_namespace", "Could not get commit history from provided namespace: {{.V0}}", params.Namespace)
 		}
+
+		if len(commits) > 0 {
+			latestRemoteID = &commits[0].CommitID // most recent commit (i.e. print all as remote)
+		}
 	} else {
 		if h.project == nil {
 			return locale.NewInputError("err_history_no_project", "A namespace was not provided and a project could not be found. Please use a project namespace or run this command in a project directory")
 		}
 
-		remoteBranch, err := model.BranchForProjectNameByName(h.project.Owner(), h.project.Name(), h.project.BranchName())
-		if err != nil {
-			return locale.WrapError(err, "err_history_remote_branch", "Could not get branch by local branch name")
-		}
-
-		remoteCommitID := remoteBranch.CommitID
 		localCommitID := h.project.CommitUUID()
 
-		lastRemoteID, err = model.CommonParent(remoteCommitID, &localCommitID)
-		if err != nil {
-			return locale.WrapError(err, "err_history_common_parent", "Could not determine common parent commit")
+		if !h.project.IsHeadless() && h.project.Name() != "" {
```

Comment:
Why have `h.project.Name() != ""`? The `IsHeadless()` should already cover that.

Reply:
From when I misunderstood how specifically "headless" was defined... fixed